### ISSUE TITLE
Error when trying to add Locale

### ISSUE THREAD

**Issue by sparkalow** [Sylius/Sylius#10303]
**Sylius version affected**: 1.4.2

**Description**  
Unable to add a new locale on fresh install. I received a php max timeout error that appears to be relate to missing "vendor/symfony/intl/Resources/data/locales/en_US.json" - en_US was the default for the fresh install.

**Steps to reproduce**  
- Install Sylius Standard via composer.
- Accept default currency. (US)
- Do not install sample data.
- Login to admin and attempt to create a new locale (/admin/locales/new)

**Comment by vvasiloi** [comment]
Wasn't able to reproduce that.
Sylius only forces the display locale for the Intl locale bundle in `Sylius\Component\Locale\Converter\LocaleConverter::convertCodeToName`, which is used in `\Sylius\Bundle\LocaleBundle\Templating\Helper\LocaleHelper::convertCodeToName` and this one is used in `\Sylius\Bundle\LocaleBundle\Twig\LocaleExtension` which registers the `sylius_locale_name` Twig filter, which by default is never called with the second parameter (the display locale), so `en` is used.
Without the Intl extension, the component is limited to `en` locale: https://symfony.com/doc/current/components/intl.html

**Comment by Zales0123** [comment]
I've been working a little bit with locales recently (while checking out https://github.com/Sylius/Sylius/pull/10240 before the merge), and everything worked fine. Can you please show your stack trace? Btw, remember you can use our [Slack](http://sylius.com/slack) or [stackoverflow](https://stackoverflow.com/questions/tagged/sylius) for solving your problem ;)

**Comment by sparkalow** [comment]
Stack trace:

```Symfony\Component\Debug\Exception\FatalErrorException:
Error: Maximum execution time of 30 seconds exceeded

  at vendor/symfony/intl/Data/Bundle/Reader/BundleEntryReader.php:112
  at Symfony\Component\Intl\Data\Bundle\Reader\BundleEntryReader->readEntry()
     (vendor/symfony/intl/Data/Provider/LocaleDataProvider.php:72)
  at Symfony\Component\Intl\ResourceBundle\LocaleBundle->getNames()
     (vendor/symfony/intl/ResourceBundle/LocaleBundle.php:56)
  at Symfony\Component\Intl\ResourceBundle\LocaleBundle->getLocaleNames()
     (vendor/sylius/sylius/src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:78)
  at Sylius\Bundle\CoreBundle\Form\Extension\LocaleTypeExtension->getAvailableLocales()
     (vendor/sylius/sylius/src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:52)
  at Sylius\Bundle\CoreBundle\Form\Extension\LocaleTypeExtension->Sylius\Bundle\CoreBundle\Form\Extension\{closure:/home/vagrant/sites/symphony.test/vendor/sylius/sylius/src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:40-57}()
     (vendor/symfony/event-dispatcher/EventDispatcher.php:212)
  at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch()
     (vendor/symfony/event-dispatcher/EventDispatcher.php:44)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch()
     (vendor/symfony/event-dispatcher/ImmutableEventDispatcher.php:33)
  at Symfony\Component\EventDispatcher\ImmutableEventDispatcher->dispatch()
     (vendor/symfony/form/Form.php:341)
  at Symfony\Component\Form\Form->setData()
     (vendor/symfony/form/Form.php:483)
  at Symfony\Component\Form\Form->initialize()
     (vendor/symfony/form/FormBuilder.php:217)
  at Symfony\Component\Form\FormBuilder->getForm()
     (vendor/symfony/form/FormFactory.php:30)
  at Symfony\Component\Form\FormFactory->create()
     (vendor/sylius/sylius/src/Sylius/Bundle/ResourceBundle/Controller/ResourceFormFactory.php:39)
  at Sylius\Bundle\ResourceBundle\Controller\ResourceFormFactory->create()
     (vendor/sylius/sylius/src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php:186)
  at Sylius\Bundle\ResourceBundle\Controller\ResourceController->createAction()
     (vendor/symfony/http-kernel/HttpKernel.php:150)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw()
     (vendor/symfony/http-kernel/HttpKernel.php:67)
  at Symfony\Component\HttpKernel\HttpKernel->handle()
     (vendor/symfony/http-kernel/Kernel.php:198)
  at App\Kernel->handle()
     (public/index.php:25)
  at {main}()
     (public/index.php:0)

**Comment by vvasiloi** [comment]
Increase the `max_execution_time` or set it to -1 in dev env and it will work.

**Comment by sparkalow** [comment]
I did increase php's max_execution_time to 10 minutes for "fun", but stopped the request after 5 minutes with no activity. I just can't wait that long :)

**Comment by vvasiloi** [comment]
The line it hands at performs an `array_merge`.
Try entering with the debugger and see what's happening there.
Like PhpStorm says:
![image](https://user-images.githubusercontent.com/7114562/56041240-eef78600-5d40-11e9-9692-de04d259a4f4.png)

**Comment by vvasiloi** [comment]
I see you are using Vagrant. Might be a filesystem and/or shared folder issue.
Try to run it on your host machine. You just need PHP, Node and a database server.

**Comment by vvasiloi** [comment]
Or just try exluding the `vendor` from shared folders.

**Comment by sparkalow** [comment]
I'm using a vagrant vm that meets the server requirements when running `php bin/console sylius:install:check-requirements`  (php 7.2, apache, mysql)

I've created a new fresh install using the instructions [here](https://docs.sylius.com/en/latest/book/installation/installation.html)

All files are on the VM now (much faster!) - no shared folder to host os.

Still errors when attempting to add a locale from the admin or visiting /admin/locales/new.  What is odd and sorry for not noticing this before is that the error changes  when loading the page between 2 possibilities. 


**Mostly this error**
```Symfony\Component\Debug\Exception\FatalErrorException:
Error: Maximum execution time of 30 seconds exceeded

  at vendor/symfony/intl/Data/Bundle/Reader/BundleEntryReader.php:88
  at Symfony\Component\Intl\Data\Bundle\Reader\BundleEntryReader->readEntry()
     (vendor/symfony/intl/Data/Provider/LocaleDataProvider.php:72)
  at Symfony\Component\Intl\ResourceBundle\LocaleBundle->getNames()
     (vendor/symfony/intl/ResourceBundle/LocaleBundle.php:56)
  at Symfony\Component\Intl\ResourceBundle\LocaleBundle->getLocaleNames()
     (vendor/sylius/sylius/src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:78)
  at Sylius\Bundle\CoreBundle\Form\Extension\LocaleTypeExtension->getAvailableLocales()
     (vendor/sylius/sylius/src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:52)
```
**But sometimes this error**
```Symfony\Component\Debug\Exception\FatalErrorException:
Error: Maximum execution time of 30 seconds exceeded

  at vendor/symfony/intl/Locale.php:71
  at Symfony\Component\Intl\Locale::getFallback()
     (vendor/symfony/intl/Data/Bundle/Reader/BundleEntryReader.php:142)
  at Symfony\Component\Intl\Data\Bundle\Reader\BundleEntryReader->readEntry()
     (vendor/symfony/intl/Data/Provider/LocaleDataProvider.php:72)
  at Symfony\Component\Intl\ResourceBundle\LocaleBundle->getNames()
     (vendor/symfony/intl/ResourceBundle/LocaleBundle.php:56)
  at Symfony\Component\Intl\ResourceBundle\LocaleBundle->getLocaleNames()
     (vendor/sylius/sylius/src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:78)
  at Sylius\Bundle\CoreBundle\Form\Extension\LocaleTypeExtension->getAvailableLocales()
     (vendor/sylius/sylius/src/Sylius/Bundle/CoreBundle/Form/Extension/LocaleTypeExtension.php:52)
  at Sylius\Bundle\CoreBundle\Form\Extension\LocaleTypeExtension->Sylius\Bundle\CoreBundle\Form\Extension\{closure}()
```

**Comment by vvasiloi** [comment]
I can't help more without reproducing the bug.
Can you export and share the image somehow?

**Comment by sparkalow** [comment]
The VM I'm using is Homestead from Laravel
https://laravel.com/docs/5.8/homestead

Here are the relevant site config bits from the homestead.yaml config file if needed.

```
 - map: vanilla-sylius.test
      to: /home/vagrant/vanilla-sylius.test/public
      type: "apache"
      php: "7.2"
```


Unfortunately, I must use a VM and cannot run on my host machine. I'm happy to try another VM setup if one is recommended. I was unable to find anything that seemed official/uptodate for Sylius.

**Comment by vvasiloi** [comment]
I will try with Laravel Homestead and in the meanwhile can you try with Docker?
```
git clone https://github.com/Sylius/Sylius-Standard.git
cd Sylius-Standard
docker-compose --version
docker-compose pull --ignore-pull-failures || true
docker-compose build --pull
docker-compose up -d
docker-compose exec php bin/console sylius:fixtures:load --no-interaction
open http://localhost/
```

**Comment by sparkalow** [comment]
Using this method I did not see the issue. I was able to add a new locale without problems. It appears this is not an issue with Sylius but related Laravel Homestead VM. 

I suspect it has to do with the nfs shared host os folder. Probably some sort of race somewhere since, refreshing the page resulted in different errors all seemingly related to reading files.

The install with Docker was a bit rocky - I understand it is still in the works currently. Also, had trouble locating the default admin credentials. Here they are as of this comment if anyone needs them - user: _sylius@example.com password: sylius_
